src/systems/optimization/constraints_system.jl

```diff
@@ -226,12 +226,15 @@ function generate_canonical_form_lhss(sys)
     lhss = subs_constants([Symbolics.canonical_form(eq).lhs for eq in constraints(sys)])
 end
 
-function get_cmap(sys::ConstraintsSystem)
+function get_cmap(sys::ConstraintsSystem, exprs = nothing)
     #Inject substitutions for constants => values
     cs = collect_constants([get_constraints(sys); get_observed(sys)]) #ctrls? what else?
     if !empty_substitutions(sys)
         cs = [cs; collect_constants(get_substitutions(sys).subs)]
     end
+    if exprs !== nothing
+        cs = [cs; collect_contants(exprs)]
```

Comment:
```suggestion
        cs = [cs; collect_constants(exprs)]
```

that must not be tested then?

Reply:
😅